xbmc/utils/GUIUtils.h

```diff
@@ -0,0 +1,45 @@
+/*
+ *  Copyright (C) 2016-2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+
+#pragma once
+
+#include "FileItem.h"
+#include "filesystem/MusicDatabaseDirectory.h"
+#include "filesystem/VirtualDirectory.h"
+#include "music/MusicDatabase.h"
+#include "music/MusicDbUrl.h"
+#include "threads/Event.h"
+#include "utils/JobManager.h"
+
+class CGUIUtils
+{
+public:
+  virtual ~CGUIUtils() = default;
+  bool PlayItem(const CFileItemPtr& itemToPlay);
+  void PlaySong(CFileItem& item);
+  bool AddItemToPlayList(const CFileItemPtr& pItem, CFileItemList& queuedItems);
+  void AddToPlayList(const CFileItemPtr& pItem, CFileItemList& queuedItems);
+  bool GetDirectory(const std::string& strDirectory, CFileItemList& items);
+  void FormatAndSort(CFileItemList& items);
+  void StartQueue();
+  void QueueComplete();
+  void QueueItem(const CFileItemPtr& itemIn,
+                 bool first = false); // false for queue, true for play next
+  void DoToastNotification(const CFileItemPtr& pItem, const int idString);
```

Comment:
This doesn't make much sense in an utils class that is supposed to be headless. Can't you just send (might have to be created) a message from your main logic to let the application know it has to display a toastnotification?
Toasts in my opinion may not be 100% GUI related. In an hypothetical headless world you might want to display a notification on the webinterface via websockets even though kodi is running headless

Reply:
Ha, this actually occurred to me last night after I had gone to bed!  As I commented earlier, I didn't originally have headless in mind when I started out, but it became apparent that this is a step in that direction as the code evolved and the necessary routines were moved out of the GUI windows.  Notifications, (or no notification at all) could easily be re-directed or not triggered by a flag of some sort.  Then whatever calls into the play or queue routines can specify where (or not) the notification should go.

Reply:
> This doesn't make much sense in an utils class that is supposed to be headless.

Yes, this functionality does not belong into this class - separation of concerns, no God classes!